CHANGELOG.md

```diff
@@ -1,5 +1,113 @@
 # TigerBeetle Changelog
 
+## 2024-07-26
+
+### Safety And Performance
+
+- [#2140](https://github.com/tigerbeetle/tigerbeetle/pull/2140)
+
+  Fix a bug where MessageBus sees block messages (due to state sync or repair) and peer_type says
+  they are always from replica 0 (since Header.Block.replica == 0 always). So, if they are being
+  sent by a non-R0 replica, it drops the messages with message from unexpected peer.
+
+- [#2137](https://github.com/tigerbeetle/tigerbeetle/pull/2137)
+
+  Ensure a prepare in the WAL is only overwritten if it belongs to a commit-quorum of checkpoints.
+
+- [#2127](https://github.com/tigerbeetle/tigerbeetle/pull/2127),
+  [#2096](https://github.com/tigerbeetle/tigerbeetle/pull/2096)
+
+  A few CI changes: run tests in CI for x86_64 macOS, add in client tests on macOS and run the
+  benchmark with `--validate` in CI.
+
+- [#2117](https://github.com/tigerbeetle/tigerbeetle/pull/2117)
+
+  TigerBeetle reserves the most significant bit of the timestamp as the tombstone flag, so indicate
+  and assert that timestamp_max is a `maxInt(u63)`.
+
+- [#2123](https://github.com/tigerbeetle/tigerbeetle/pull/2123),
+  [#2125](https://github.com/tigerbeetle/tigerbeetle/pull/2125)
+
+  Internally, TigerBeetle uses AEGIS-128L for checksumming - hardware AES is a prerequisite for
+  performance. Due to a build system bug, releases being built with a specified (`-Dtarget=`) target
+  would only be built with baseline CPU features, and thus use the software AES implementation.
+
+  Enforce at comptime that hardware acceleration is available, fix the build system bug, log
+  checksum performance on our [devhub](https://tigerbeetle.github.io/tigerbeetle/) and build client
+  libraries with hardware acceleration too.
+
+- [#2139](https://github.com/tigerbeetle/tigerbeetle/pull/2139)
+
+  TigerBeetle would wait until all repairable headers are fixed before trying to commits prepares,
+  but if all the headers after the checkpoint are present then we can start committing even if
+  some headers from before the checkpoint are missing.
+
+- [#2141](https://github.com/tigerbeetle/tigerbeetle/pull/2141)
+
+  Clarify that the order of replicas in `--addresses` is important.
```

Comment:
Again, highlight the _why_ of why it's important. e.g. Otherwise replica routing will be impacted, degrading performance significantly.